packages/sbom-tools/src/webpack-dependencies-plugin.spec.ts

```diff
@@ -0,0 +1,301 @@
+import { expect } from 'chai';
+import path from 'path';
+import fs from 'fs';
+import webpack from 'webpack';
+import type { WebpackDependenciesPluginOptions } from './webpack-dependencies-plugin';
+import { WebpackDependenciesPlugin } from './webpack-dependencies-plugin';
+import type { Package } from './get-package-info';
+import { cleanup, setupTempDir, toCleanup } from '../test/helpers';
+
+function runPlugin(
+  structure: Record<string, string>,
+  options: WebpackDependenciesPluginOptions
+): Promise<Package[]> {
+  const contextPath = setupTempDir(structure);
+
+  const outputPath =
+    options.outputFilename ?? path.join(contextPath, 'dependencies.json');
+  toCleanup.push(outputPath);
+
+  return new Promise((resolve, reject) => {
+    const plugin = new WebpackDependenciesPlugin(options);
+    const config = fs
+      .readdirSync(path.join(contextPath, './src'))
+      .filter(
+        (filename) => filename.startsWith('input') && filename.endsWith('.js')
+      )
+      .map((filename, i) => ({
+        context: contextPath,
+        entry: path.resolve(contextPath, 'src', filename),
+        output: {
+          path: path.resolve(contextPath, 'dist'),
+          filename: `output-${i}.js`,
+        },
+        plugins: [plugin],
+      }));
+
+    webpack(config, (err, stats) => {
```

Comment:
This looks like something worth `util.promisify()`ing and then just using it with `await`